superset/commands/base.py

```diff
@@ -0,0 +1,39 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from abc import ABC, abstractmethod
+
+
+class BaseCommand(ABC):
+    """
+        Base class for all Command like Superset Logic objects
+    """
+
+    @abstractmethod
+    def run(self):
```

Comment:
Slight tweak here: 

1. Make `run()` non-abstract
2. In `run()`, call `validate()`
3. Define / implement a method named something like `do_run()` which is abstract, which subclasses will implement. 

This ensures that validate() is always called, and allows us to place some boiler-plate logic in `run()` (think transaction handling)

Reply:
I disagree. I'd like for us to keep `run` abstract and not make assumptions about the resulting command implementation. So far commands have wrapped single database operations, but there's no guarantee that this will always be the case. Commands may not interact with the metadata database at all, or they may require multiple transactions. Not all commands may need a validate() step. I do not want the base command to be prescriptive.

Reply:
If they don't need a validate(), they can just not implement the abstract `validate()` from the base class, in which case validate() performs a no-op (which is currently the case). If we aren't calling `validate()` from outside the command, or from the base class, then there's no point in having it defined here.

Reply:
I'm agnostic about `validate()` as a top-level entrypoint or not. Either way is fine by me.